app/models/setting/general.rb

```diff
@@ -29,6 +29,8 @@ def self.default_settings
       self.set('instance_id', N_("Foreman instance ID, uniquely identifies this Foreman instance."), 'uuid', N_('Foreman UUID'), Foreman.uuid),
       self.set('default_locale', N_("Language to use for new users"), nil, N_('Default language'), nil, { :collection => Proc.new { locales } }),
       self.set('default_timezone', N_("Timezone to use for new users"), nil, N_('Default timezone'), nil, { :collection => Proc.new { timezones } }),
+      self.set('enable_facts_storage', N_("Store Fact Values"), FactValue.unscoped.any?, N_('Store fact values in the database including historical records')),
```

Comment:
I think the default value shouldn't be dynamic, we detect whether user changed it based on it. This could switch by deleting hosts (and it's facts). Similarly with reports.

I like the idea of configuring this based on present data. If we let's say default to `false`, we could still add a migration that would set this to `true` in case there's at least one fact.

The very same for reports below.

Reply:
Migration added.